cmd/gcp-controller-manager/app/node_annotater_test.go

```diff
@@ -192,3 +195,93 @@ func TestExtrackKubeLabels(t *testing.T) {
 		})
 	}
 }
+
+func TestManageNodeTerminationTaint(t *testing.T) {
+	cs := map[string]struct {
+		annotation map[string]string
+		update     bool
+		taints     []core.Taint
+		out        []core.Taint
+	}{
+		"no annotation needs update": {
+			annotation: nil,
+			taints: []core.Taint{
+				{
+					Key:    "cloud.google.com/impending-node-termination",
+					Effect: core.TaintEffectNoSchedule,
+				},
+			},
+			update: true,
+			out:    []core.Taint{},
+		},
+		"no annotation no update": {
+			annotation: nil,
+			taints:     []core.Taint{},
+			update:     false,
+			out:        []core.Taint{},
+		},
+		"annotation true": {
+			annotation: map[string]string{"cloud.google.com/impending-node-termination": "true"},
+			taints:     []core.Taint{},
+			update:     true,
+			out: []core.Taint{
+				{
+					Key:    "cloud.google.com/impending-node-termination",
+					Effect: core.TaintEffectNoSchedule,
+				},
+			},
+		},
+		"annotation true no update": {
+			annotation: map[string]string{"cloud.google.com/impending-node-termination": "true"},
+			taints: []core.Taint{
+				{
+					Key:    "cloud.google.com/impending-node-termination",
+					Effect: core.TaintEffectNoSchedule,
+				},
+			},
+			update: false,
+			out: []core.Taint{
+				{
+					Key:    "cloud.google.com/impending-node-termination",
+					Effect: core.TaintEffectNoSchedule,
+				},
+			},
+		},
+		"annotation false": {
+			annotation: map[string]string{"cloud.google.com/impending-node-termination": "false"},
+			taints: []core.Taint{
+				{
+					Key:    "cloud.google.com/impending-node-termination",
+					Effect: core.TaintEffectNoSchedule,
+				},
+			},
+			update: true,
+			out:    []core.Taint{},
+		},
+	}
+
+	for name, c := range cs {
+		t.Run(name, func(t *testing.T) {
+			node := &core.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: c.annotation,
+				},
+				Spec: core.NodeSpec{
+					Taints: c.taints,
+				},
+			}
+
+			updated := handleNodeTerminations(node)
+			if updated != c.update {
+				t.Fatalf("test case %q: expected update %v, got: %v", name, c.update, updated)
```

Comment:
remove `test case %q: ` from all error messages, it's already present in subtest name